cuqi/samples/_samples.py

```diff
@@ -873,3 +873,6 @@ def plot_violin(self, variable_indices=None, **kwargs):
         ax =  arviz.plot_violin(datadict, **kwargs)
 
         return ax
+
+    def __repr__(self) -> str:
```

Comment:
Displaying the shape is a really nice addition, I suggest we can take this a step further to mention which dimension is the number of samples and which is the dim of the samples. Hence, the properties: `Ns` for the number of samples and `s.geometry.par_dim` for the dimension of samples can be used instead of the shape

Reply:
Done

Reply:
While we are at it, why not list the geometry?

Reply:
Done